### ISSUE TITLE
Misleading Docs since Velero Supervisor Services Not Visible

### ISSUE THREAD

**Issue by kacole2** [vmware-tanzu/velero-plugin-for-vsphere#336]
**Describe the bug**

I'm trying to do some testing on backup and restore of a Supervisor Cluster.

Looking at the documentation for [Installing Velero on Supervisor cluster](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/docs/velero-vsphere-operator-user-manual.md#installing-velero-on-supervisor-cluster), it shows there should a service to enable. I just installed a fresh vSphere with Tanzu U2 environment and do not have any services available to install. I'm not using NSX-T nor VCF. This is a simple installation using the Distributed Switch and NSX-ALB. 

If this behavior is expected, the documents should be updated to reflect the platform or criteria required to have these services enabled. 

**To Reproduce**

vSphere with Tanzu using the Virtual Distributed Switch with NSX-ALB. Do not use NSX-T or VCF.

**Expected behavior**

The documentation should reflect the requirements to have Supervisor Services. If these Services are not available, where is the documentation to do a backup and restoration of a Supervisor Cluster when these services cannot be enabled?

**Screenshots**

![Image of vSphere with Tanzu](https://i.ibb.co/7ps2C02/Screen-Shot-2021-04-08-at-10-46-20-AM.png)

**Comment by lintongj** [comment]
We put [the requirement of NSX-T](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/README.md#important-note-on-vsphere-with-tanzu-networking-requirements) on the landing page. Please double check.

**Comment by xing-yang** [comment]
Maybe we should add a link to pre-requisite before step 1 in this section to make it more clear:
https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/docs/velero-vsphere-operator-user-manual.md#installing-velero-on-supervisor-cluster

**Comment by kacole2** [comment]
Thanks for the pointer @lintongj. I completely overlooked it and went straight to the installation. Does this solution rely on vSphere Pods to work? Have there been discussions on how this will work for the majority of installations that are going to be based on vSphere Distributed Networking? Thanks!

**Comment by lintongj** [comment]
>Does this solution rely on vSphere Pods to work? 

Yes. Currently, velero-plugin-for-vsphere require Pod VM to work on vSphere with Tanzu.

>Have there been discussions on how this will work for the majority of installations that are going to be based on vSphere Distributed Networking?

Yes. The support to enable vSphere Pod on native vSphere distributed switch networking is underway. It is a vSphere feature. I am not sure the timeline though. 

>Should vanilla kubernetes be used in this case? Will that successfully back and restore a Supervisor Cluster?

You can definitely deploy vanilla kubernetes cluster on your current vSphere environment, outside the Supervisor Cluster. But it cannot be used to backup and restore a Supervisor Cluster.

**Comment by lintongj** [comment]
> Maybe we should add a link to pre-requisite before step 1 in this section to make it more clear:
> https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/docs/velero-vsphere-operator-user-manual.md#installing-velero-on-supervisor-cluster

I think we can add the NSX-T requirement into [the pre-requisite section](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/main/docs/supervisor.md#prerequisites).

**Comment by kacole2** [comment]
I also feel there might be a gap in the messaging. We mention Velero support for backing up the Supervisor Cluster in Tanzu Basic. We should note that this is only supported when NSX-T is also a part of the solution.

**Comment by xing-yang** [comment]
> I also feel there might be a gap in the messaging. We mention Velero support for backing up the Supervisor Cluster in Tanzu Basic. We should note that this is only supported when NSX-T is also a part of the solution.

By default, Tanzu Basic does not support Pod VM so Velero Plugin for vSphere will not work, but it is possible for customer to buy NSX and Tanzu Basic to get PodVM which allows them to use vSphere Plugin to take VolumeSnapshots.

If there's an official document that says "Velero support for backing up the Supervisor Cluster in Tanzu Basic", we should re-visit and fix it.

**Comment by kacole2** [comment]
This goes beyond Tanzu Basic because every Tanzu Edition does not come with NSX-T (including Advanced). It's an additional add-on. We also mention support for Velero to backup the Tanzu Kubernetes Clusters/Workload/Guest Clusters with Tanzu Standard and Advanced. What is the proper method for backing up these clusters that are not using NSX-T? Should the vSphere driver be used at all or should it be the generic Velero+Restic backup method?

There are internal confluence pages that need to call this out. I adjusted the Tanzu Edition comparison matrix for backing up Supervisor Clusters to say there is no possible solution unless you have NSX-T.

**Comment by xing-yang** [comment]
> This goes beyond Tanzu Basic because every Tanzu Edition does not come with NSX-T (including Advanced). It's an additional add-on. We also mention support for Velero to backup the Tanzu Kubernetes Clusters/Workload/Guest Clusters with Tanzu Standard and Advanced. What is the proper method for backing up these clusters that are not using NSX-T? Should the vSphere driver be used at all or should it be the generic Velero+Restic backup method?

> There are internal confluence pages that need to call this out. I adjusted the Tanzu Edition comparison matrix for backing up Supervisor Clusters to say there is no possible solution unless you have NSX-T.

On Tanzu Edition that does not come with NSX-T, Velero+Restic should be used.  @dsu-igeek has put together an internal confluence page that summarizes what is supported on different flavors.  We should have an official doc with this information so that we can reference it from the README doc in this github repo.

**Comment by Aqualie** [comment]
Are there any plans in the future to provide support for NSX-ALB? I was planning on levering Velero for migrating my workload that is presently running with NSX-T to NSX-ALB. However due to NSX-ALB not being supported I wouldn't be able to perform a restore as there is not support for NSX-ALB..
